jena-core/src/main/java/org/apache/jena/util/PrefixMappingUtils.java

```diff
@@ -0,0 +1,362 @@
+            if ( pmap.size() == inUseURIs.size() )
+                break ;
+        }
+        return inUseURIs ;
+    }
+
+    private static void processFull(Trie<String> trie, Set<String> prefixesInUse, Node node) {
+        if ( ! node.isURI() )
+            return ;
+        String uri = node.getURI() ;
+        // Shorten to "/" or "#" or ":"
+        String pref = uri ;
+        
+        // Get all under the pref
+        List<String> hits = trie.partialSearch(pref) ;
+        if ( hits == null || hits.isEmpty() )
+            return ;
+        //System.out.println(pref+" => ("+hits.size()+")"+hits) ;
+        prefixesInUse.addAll(hits) ;
+    }
+    
+    // -------------------------------------------
+
+    /** Assume that prefixes:localName are in the normal places (after/ or #).
+     * i.e. node.getNameSpace() ; makes sense.
+     * @param m
+     */
+    private static Set<String> splitMethod(Model m) {
+        /* Method:
+         * 
+         * For each URI, split in in the usual place, after "/" or "#" for http URIs, and
+         * after the last ":" for URNs, then see if that is a declared prefix.
+         * 
+         * Exit early if every prefix is accounted for. 
+         */
+        
+        PrefixMapping prefixMapping = m ;
+        // Map prefix -> URI.
+        Map<String, String> pmap = prefixMapping.getNsPrefixMap() ;    
+        
+        // All URIs used as prefixes in the prefix mapping.
+        Set<String> prefixURIs = new HashSet<>(pmap.values()) ;             
+        
+        // Prefix URIs used.
+        Set<String> inUsePrefixURIs = new HashSet<>() ;
+        
+        Iterator<Triple> iter = m.getGraph().find(null, null, null) ;
+        while(iter.hasNext()) {
+            Triple triple = iter.next() ;
+            processBySplit(prefixURIs, inUsePrefixURIs, triple.getSubject()) ; 
+            processBySplit(prefixURIs, inUsePrefixURIs, triple.getPredicate()) ;
+            processBySplit(prefixURIs, inUsePrefixURIs, triple.getObject()) ; 
+            if ( inUsePrefixURIs.size() == prefixURIs.size() )
+                // Fast exit.
+                break ;
+        }
+        return inUsePrefixURIs ;
+    }
+        
+    private static void processBySplit(Collection<String> prefixURIs, Collection<String> inUse, Node node) {
+        if ( ! node.isURI() )
+            return ;
+        String uri = node.getURI() ;
+        
+        int idx = SplitIRI.splitpoint(uri) ;
+        if ( idx < 0 )
+            return ;
+        String prefixUri = uri.substring(0,idx) ;
+        String localname = uri.substring(idx) ;
+        
+        if ( prefixURIs.contains(prefixUri) )
+            inUse.add(prefixUri) ;
+//        String ns = node.getNameSpace() ;
+//        if ( ns == null )
+//            return ;
+//        if ( prefixURIs.contains(ns) )
+//            inUse.add(ns) ;
+    }
+    
+    private static void print(Set<String> inUsePrefixURIs, PrefixMapping prefixMapping) {    
```

Comment:
Should we leave this in? It looks like debugging code?

Reply:
The algorithm took a while to get right and being about to print the state was useful.  I'll add a comment that's it for development assistance. Maybe it will never needed.

Reply:
The algorithm took a while to get right and being about to print the state was useful.  I'll add a comment that's it for development assistance. Maybe it will never needed.